packages/core-db/test/app/ethereum/ethereum-block-processor.spec.ts

```diff
@@ -37,6 +37,9 @@ describe('Block Subscription', () => {
 
     tokenContract = await deployTokenContract(ownerWallet, initialSupply)
 
+    // Wait for deploy block to be mined
+    await sleep(25)
```

Comment:
This race condition could randomly fail on CI if it runs on a busy machine. I know `sleep` is an an easy fix, but you could do something like:

```js
let seen = false;
provider.on('block', () => {
  seen = true
})

let ticks = 0
await new Promise((resolve, reject) => {
  setInterval(() => {
    if (ticks > 5)
      return reject()
    if (seen)
      return resolve()
    ticks++
  }, 10)
})
```

Assuming that the provider here has a mocked out event emitter set up with the block event.

See bcoin helper function: https://github.com/bcoin-org/bcoin/blob/f1abba52eda16d3981039651d0e0407e0c7fc778/test/util/common.js#L105

Reply:
Ahh, my bad. Going to remove the sleep because I don't actually think it's necessary 😆 

FWIW, unfortunately issues with truly async code cannot be prevented (without mocking to make it sync), but what you propose does have the nice benefits of not sleeping the full time period if it's ready _before_ that and failing right at the point of the problem if it does fail. But yeah, this errors just as `sleep(50)` in the worst case.